### ISSUE TITLE
[B3.1] Create shared/autograd/schedulers.mojo with StepLR and ExponentialLR

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ml-odyssey#2303]
## Overview
Create learning rate schedulers for adaptive learning rate decay during training.

## Location
- Create new `shared/autograd/schedulers.mojo`

### StepLR
Decay learning rate by gamma every step_size epochs:
```
lr = base_lr * gamma^(epoch // step_size)
```

### ExponentialLR
Decay learning rate by gamma every epoch:
```
lr = base_lr * gamma^epoch
```

## Deliverables
- [ ] Create `shared/autograd/schedulers.mojo`
- [ ] Implement LRScheduler base trait
- [ ] Implement StepLR scheduler:
  - step_size: epochs between decay
  - gamma: multiplicative decay factor
- [ ] Implement ExponentialLR scheduler:
  - gamma: multiplicative decay factor
- [ ] Export from `shared/autograd/__init__.mojo`

## Success Criteria
- Schedulers integrate with optimizer learning rates
- Correct learning rate decay behavior
- CI passes

## Dependencies
- None (first in B3 chain)

## Track
Track B: Missing Features (B3.1 of 8)
